frontend/src/metabase/static-viz/components/LineAreaBarChart/utils/series.unit.spec.ts

```diff
@@ -0,0 +1,401 @@
+
+    expect(seriesWithColors).toEqual([]);
+  });
+
+  describe("Series without ones or less dimensions", () => {
+    const multipleSeries: SeriesWithOneOrLessDimensions[][] = [
+      [
+        {
+          name: "Count",
+          cardName: "Bar chart",
+          yAxisPosition: "left",
+          type: "bar",
+          data: [
+            ["Doohickey", 3976],
+            ["Gadget", 4939],
+            ["Gizmo", 4784],
+            ["Widget", 5061],
+          ],
+          column: {
+            name: "count",
+            source: "aggregation",
+            display_name: "Count",
+          },
+        },
+      ],
+    ];
+
+    it("should assign colors given series", () => {
+      const seriesWithColors = getSeriesWithColors(
+        multipleSeries,
+        settings,
+        getPalette({}),
+      );
+
+      const expectedSeries = [
+        [
+          {
+            color: "#509EE3", // brand color
+            name: expect.anything(),
+            cardName: expect.anything(),
+            yAxisPosition: expect.anything(),
+            type: expect.anything(),
+            data: expect.anything(),
+            column: expect.anything(),
+          },
+        ],
+      ];
+
+      expect(seriesWithColors).toEqual(expectedSeries);
+    });
+
+    it("should assign colors from whitelabel colors", () => {
+      const seriesWithColors = getSeriesWithColors(
+        multipleSeries,
+        settings,
+        getPalette({ brand: "#123456", summarize: "#ffffff" }),
+      );
+
+      const expectedSeries = [
+        [
+          {
+            color: "#123456", // whitelabel color
+            name: expect.anything(),
+            cardName: expect.anything(),
+            yAxisPosition: expect.anything(),
+            type: expect.anything(),
+            data: expect.anything(),
+            column: expect.anything(),
+          },
+        ],
+      ];
+
+      expect(seriesWithColors).toEqual(expectedSeries);
+    });
+
+    it("it should assign colors from column colors", () => {
+      const seriesWithColors = getSeriesWithColors(
+        multipleSeries,
+        merge(settings, {
+          series_settings: {
+            count: {
+              color: "#987654",
+            },
+          },
+        }),
+        getPalette({ brand: "#123456", summarize: "#ffffff" }),
+      );
+
+      const expectedSeries = [
+        [
+          {
+            color: "#987654", // column color
+            name: expect.anything(),
+            cardName: expect.anything(),
+            yAxisPosition: expect.anything(),
+            type: expect.anything(),
+            data: expect.anything(),
+            column: expect.anything(),
+          },
+        ],
+      ];
+
+      expect(seriesWithColors).toEqual(expectedSeries);
+    });
+  });
+
+  describe("Series with preferred colors", () => {
+    const multipleSeries: SeriesWithOneOrLessDimensions[][] = [
```

Comment:
Sampled from this chart
![image](https://user-images.githubusercontent.com/1937582/201139555-96fdbaad-b929-4855-b8a9-da09665dd4c4.png)